src/main/java/org/jabref/gui/keyboard/EmacsKeyBindings.java

```diff
@@ -0,0 +1,103 @@
+package org.jabref.gui.keyboard;
+
+import java.util.Optional;
+
+import javafx.scene.Scene;
+import javafx.scene.control.TextInputControl;
+import javafx.scene.input.KeyEvent;
+
+import org.jabref.Globals;
+import org.jabref.logic.util.strings.EmacsStringManipulator;
+import org.jabref.model.util.ResultingEmacsState;
+import org.jabref.preferences.JabRefPreferences;
+
+public class EmacsKeyBindings {
+
+    public static void executeEmacs(Scene scene, KeyEvent event) {
+        boolean EmacsFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS);
+        if (EmacsFlag && scene.focusOwnerProperty().get() instanceof TextInputControl) {
+            boolean CAFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS_REBIND_CA);
+            boolean CFFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS_REBIND_CF);
+            boolean CNFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS_REBIND_CN);
+            boolean AUFlag = Globals.prefs.getBoolean(JabRefPreferences.EDITOR_EMACS_KEYBINDINGS_REBIND_AU);
+
+            KeyBindingRepository keyBindingRepository = Globals.getKeyPrefs();
+            TextInputControl focusedTextField = (TextInputControl) scene.focusOwnerProperty().get();
+            Optional<KeyBinding> keyBinding = keyBindingRepository.mapToKeyBinding(event);
+            if (keyBinding.isPresent()) {
```

Comment:
Either you use the lamdba with keyBindinding.ifPresent(keybinding->...) or you extract the keyBinding.get() to a variable in the first if and reuse that. Less code duplication.
And second, you are comparing enum values, they can be compared with `==` no need for equals
And here is an explanation why: https://stackoverflow.com/a/2937561